core/trino-main/src/test/java/io/trino/execution/TestDropColumnTask.java

```diff
@@ -0,0 +1,184 @@
+    }
+
+    @Test
+    public void testDroColumnNotExistingTable()
+    {
+        QualifiedObjectName tableName = qualifiedObjectName("not_existing_table");
+
+        assertTrinoExceptionThrownBy(() -> getFutureValue(executeDropColumn(asQualifiedName(tableName), QualifiedName.of("test"), false, false)))
+                .hasErrorCode(TABLE_NOT_FOUND)
+                .hasMessageContaining("Table '%s' does not exist", tableName);
+    }
+
+    @Test
+    public void testDropColumnNotExistingTableIfExists()
+    {
+        QualifiedName tableName = qualifiedName("not_existing_table");
+
+        getFutureValue(executeDropColumn(tableName, QualifiedName.of("test"), true, false));
+        // no exception
+    }
+
+    @Test
+    public void testDroColumnNotExistingColumn()
+    {
+        QualifiedObjectName tableName = qualifiedObjectName("existing_table");
+        metadata.createTable(testSession, TEST_CATALOG_NAME, simpleTable(tableName), false);
+
+        assertTrinoExceptionThrownBy(() -> getFutureValue(executeDropColumn(asQualifiedName(tableName), QualifiedName.of("not_existing"), false, false)))
+                .hasErrorCode(COLUMN_NOT_FOUND)
+                .hasMessageContaining("Column 'not_existing' does not exist");
+    }
+
+    @Test
+    public void testDropColumnIfExists()
+    {
+        QualifiedObjectName tableName = qualifiedObjectName("existing_table");
+        metadata.createTable(testSession, TEST_CATALOG_NAME, simpleTable(tableName), false);
+        TableHandle table = metadata.getTableHandle(testSession, tableName).get();
+
+        getFutureValue(executeDropColumn(asQualifiedName(tableName), QualifiedName.of("c"), false, true));
+        assertThat(metadata.getTableMetadata(testSession, table).getColumns())
+                .isEqualTo(ImmutableList.of(new ColumnMetadata("a", BIGINT), new ColumnMetadata("b", BIGINT)));
+    }
+
+    @Test
+    public void testUnsupportedDropDuplicatedField()
+    {
+        QualifiedObjectName tableName = qualifiedObjectName("existing_table");
+        metadata.createTable(testSession, TEST_CATALOG_NAME, rowTable(tableName, new Field(Optional.of("a"), BIGINT), new Field(Optional.of("a"), BIGINT)), false);
+        TableHandle table = metadata.getTableHandle(testSession, tableName).get();
+        assertThat(metadata.getTableMetadata(testSession, table).getColumns())
+                .isEqualTo(ImmutableList.of(new ColumnMetadata("col", RowType.rowType(
+                        new Field(Optional.of("a"), BIGINT), new Field(Optional.of("a"), BIGINT)))));
+
+        assertTrinoExceptionThrownBy(() -> getFutureValue(executeDropColumn(asQualifiedName(tableName), QualifiedName.of("col", "a"), false, false)))
+                .hasErrorCode(COLUMN_NOT_FOUND)
+                .hasMessageContaining("Field path [a] within row(a bigint, a bigint) is ambiguous");
+    }
+
+    @Test
+    public void testUnsupportedDropOnlyField()
```

Comment:
For my understanding does removing a field from a row types comes under `DROP COLUMN` or `SET COLUMN TYPE` or both ?

Reply:
Both.

Reply:
So which is a correct way to do ? Both or ?

Reply:
It depends on user's choice and connector's implementation. Some connectors may implement either one.